store.go

```diff
@@ -3843,3 +3853,41 @@ func (s *store) MultiList(options MultiListOptions) (MultiListResult, error) {
 	}
 	return out, nil
 }
+
+// Dedup deduplicates layers in the store.
+func (s *store) Dedup(req DedupArgs) (drivers.DedupResult, error) {
+	imgs, err := s.Images()
+	if err != nil {
+		return drivers.DedupResult{}, err
+	}
+	var topLayers []string
+	for _, i := range imgs {
+		topLayers = append(topLayers, i.TopLayer)
+		topLayers = append(topLayers, i.MappedTopLayers...)
+	}
```

Comment:
we should not deduplicate writeable layers, as that can slow down containers at runtime.  If a container has its own copy of a file, and we deduplicate it, then the next time the container tries to write to it, the file system needs to create a copy.  The current logic instead has no such disadvantages because read-only layers referenced by an image can't change